### PR TITLE
deps: migrate from bower to yarn

### LINKED ISSUE #2164: Consider using yarn/npm to replace bower
When you run any newer version of bower, it asserts that you should try yarn or webpack for new project.  The commit which introduced it is https://github.com/bower/bower.github.io/commit/12566e5b777ef4c93c525cf3d2fa35ebc9f4d2e5.  Some people [consider yarn to be a viable replacement for bower](https://stackoverflow.com/questions/43562645/can-yarn-be-considered-a-viable-option-as-a-replacement-for-bower-and-npm).  Unfortunately, yarn officially removed bower support (https://github.com/yarnpkg/yarn/pull/1441), so it would be a little bit of work to combine them all together.

Still, it seems like bower is on its way out and we might consider `npm` or `yarn` as a viable alternative before we migrate to something extreme like webpack.

### PR DESCRIPTION
**Pull request by jniles**
BREAKING CHANGE: removes bower from installation process in favor of yarn.  Bower officially recommends moving to yarn/webpack/npm and so we want to get ahead of them breaking our builds by going dark.  It also simplifies the installation process for new contributors.

Closes #2164.

**TODO**
 - [x] Verify that tests still pass.
 - [x] Verify that application functions as expected.
 - [x] Remove references to `bower` from documentation
 
I might also upgrade gulp to v4, since that will also be a breaking change.

### CODE REVIEW

**Comment by jniles** [comment]
Tests pass! :dancer:

**Comment by jniles** [comment]
@mbayopanda, @lomamech, @jeremielodi what do you think of this change?

File: .travis.yml [hunk 1]
```diff
@@ -17,7 +17,8 @@ node_js:
 before_script:
   - sudo /etc/init.d/mysql restart
   - export DISPLAY=:99.0
-  - ./sh/dependencies.sh
+  - yarn build
```
Comment (mbayopanda) [inline .travis.yml:20]: 👍

**Review by mbayopanda — APPROVED** [review]
The build process works perfectly on my windows machine.

LGTM 👍 

bors r+

**Pull request merged**